tests/test_core_network.py

```diff
@@ -500,12 +516,36 @@ def test_simplifing_puma_network_results_in_correct_record_of_removed_links_and_
     assert report['schedule']['schedule_level']['is_valid_schedule']
 
 
-def test_simplified_network_saves_to_correct_dtds(tmpdir, network_dtd, schedule_dtd):
-    n = read.read_matsim(path_to_network=puma_network_test_file, epsg='epsg:27700',
-                         path_to_schedule=puma_schedule_test_file)
+def test_simplify_does_not_oversimplify_PT_endpoints(
+        puma_network_pre_simplify, puma_network_post_simplify):
+    n = puma_network_pre_simplify
+
+    stops_at_risk = ['5221390681543854913', '5221390302070799085', '5221390323679791901']
+    for s in stops_at_risk:
+        assert n.link(n.schedule.stop(s).linkRefId)['length'] == 1
+
+    deleted_self_loops, n = puma_network_post_simplify
+
+    for s in stops_at_risk:
+        assert n.link(n.schedule.stop(s).linkRefId)['length'] == 1
+
+
+def test_simplify_removes_loops_by_default_but_keeps_pt_stop_loops(
+        puma_network_pre_simplify, puma_network_post_simplify):
+    n = puma_network_pre_simplify
+
+    assert n.schedule.stop('5221390681543854913').linkRefId == '5221390681543854913_5221390681543854913'
+
+    deleted_self_loops, n = puma_network_post_simplify
+
+    assert n.schedule.stop('5221390681543854913').linkRefId == '5221390681543854913_5221390681543854913'
+    assert n.has_link('5221390681543854913_5221390681543854913')
+    assert n.link('5221390681543854913_5221390681543854913')['from'] == '5221390681543854913'
+    assert n.link('5221390681543854913_5221390681543854913')['to'] == '5221390681543854913'
 
-    n.simplify()
 
+def test_simplified_network_saves_to_correct_dtds(tmpdir, network_dtd, schedule_dtd, puma_network_post_simplify):
+    deleted_self_loops, n = puma_network_post_simplify
     n.write_to_matsim(tmpdir)
```

Comment:
We're kind of testing two things at once here - network simplification, _and_ network serialisation. This test could therefore presumably fail if either the serialisation code was broken, or there was something funky going on in the simplification that meant it couldn't be serialised properly. We want the number of things that can cause our tests to fail to be as small as possible. Is there a way to reduce the number of things that can make this test fail?